### ISSUE TITLE
[HIGH][DOC-003,DOC-004] Add comprehensive API documentation and examples for quantization methods

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#1917]
## Problem

Incomplete API documentation for quantization methods (to_mxfp4, to_nvfp4, from_mxfp4, from_nvfp4). Missing examples, error cases, and edge case guidance.

## Impact

**HIGH**: Developers will misuse API, lose data, get wrong results

**Risk**: Silent data corruption from misunderstanding padding behavior, dtypes, shape requirements

## Locations

- \`shared/core/extensor.mojo:1459-1478\` (to_mxfp4 docs)
- \`shared/core/extensor.mojo:1659-1678\` (to_nvfp4 docs)

## FIXME Markers

Both methods marked with:
\`\`\`
# FIXME (DOC-003, DOC-004): Incomplete API documentation
\`\`\`

### 1. Non-Aligned Tensor Examples (DOC-003)
**Current**: Only aligned tensor examples
**Needed**: Examples with padding behavior

\`\`\`mojo
fn to_mxfp4(self) raises -> ExTensor:
    """Convert tensor to MXFP4 format.
    
    ...
    
    Examples:
        # Aligned tensor (32 elements, 1 block, no padding)
        >>> var t = zeros(List[Int](32), DType.float32)
        >>> var mxfp4 = t.to_mxfp4()  # Exact fit, no padding
        
        # Non-aligned tensor (33 elements, needs padding to 64)
        >>> var t2 = zeros(List[Int](33), DType.float32)
        >>> var mxfp4_2 = t2.to_mxfp4()  # Padded to 64 elements (2 blocks)
        >>> print(mxfp4_2.numel())  # 64, not 33!
        
        # Multi-dimensional tensor
        >>> var t3 = zeros(List[Int](5, 7), DType.float32)  # 35 elements
        >>> var mxfp4_3 = t3.to_mxfp4()  # Padded to 64 (2 blocks)
        >>> # Shape preserved: (5, 7) but data padded
        
        # Round-trip with dimension loss (CAUTION!)
        >>> var original = zeros(List[Int](33), DType.float32)
        >>> var quantized = original.to_mxfp4()
        >>> var restored = quantized.from_mxfp4()
        >>> print(restored.numel())  # 64, not 33! (see DATA-001)
    """
\`\`\`

### 2. Error Case Examples (DOC-004)
**Current**: No error documentation
**Needed**: All failure modes documented

\`\`\`mojo
    Raises:
        Error: If tensor is empty (numel == 0)
        Error: If tensor dtype is not float32
        Error: If memory allocation fails (OOM)
        Error: If tensor contains NaN/Inf (optional - may clamp instead)
        
    Notes on Padding:
        - MXFP4 requires 32-element blocks
        - Tensors padded to next multiple of 32
        - Padding filled with zeros
        - Original size NOT preserved (see QuantizedTensor wrapper)
        - Shape dimensions preserved, but data size changes
        
        Example padding sizes:
        - 1-32 elements → 32 (1 block)
        - 33-64 elements → 64 (2 blocks)
        - 65-96 elements → 96 (3 blocks)
\`\`\`

### 3. ML Workflow Examples (DOC-003)
**Current**: No context for ML usage
**Needed**: Complete workflows

\`\`\`mojo
    ML Workflow Examples:
        # Quantize gradients for communication
        >>> var gradients = compute_gradients(model, batch)
        >>> var quantized_grads = gradients.to_mxfp4()
        >>> send_to_workers(quantized_grads)  # 4× smaller!
        
        # Quantize weights for inference
        >>> var weights = model.get_weights()
        >>> var mxfp4_weights = weights.to_mxfp4()
        >>> save_checkpoint(mxfp4_weights)  # 8× smaller checkpoint
        
        # Mixed-precision training
        >>> var fp32_activations = model.forward(x)
        >>> var mxfp4_grads = compute_grads().to_mxfp4()  # Low-precision grads
        >>> optimizer.step(mxfp4_grads.from_mxfp4())  # Restore for update
\`\`\`

## Estimated Effort

2-3 hours per method (4 methods total)
**Total**: 8-12 hours

## Success Criteria

- [ ] Non-aligned tensor examples added
- [ ] Error cases fully documented
- [ ] Padding behavior explained with examples
- [ ] Multi-dimensional tensor handling documented
- [ ] Round-trip limitations explained (link to DATA-001)
- [ ] ML workflow examples provided
- [ ] All failure modes documented
- [ ] OOM handling documented
- [ ] NaN/Inf behavior documented

## References

- FIXME_SUMMARY.md: Lines 97-101, 218-225
- Related: DATA-001 (padding data loss), DATA-002 (size tracking)

## Priority

**P1 HIGH** - Prevents API misuse
